File: final_dataset_A/generated_code/claude/java/DownloadWebPage-claude.java
import java.io.*;
import java.net.HttpURLConnection;
import java.net.URL;
import java.util.Scanner;

public class DownloadWebPage {

    public static void downloadPage(String urlString) throws Exception {
        URL url = new URL(urlString);
        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
        conn.setRequestMethod("GET");

        BufferedReader reader = new BufferedReader(new InputStreamReader(conn.getInputStream()));
        FileWriter writer = new FileWriter("downloaded_page.html");

        String line;
        while ((line = reader.readLine()) != null) {
            writer.write(line + "\n");
        }
        reader.close();
        writer.close();
        System.out.println("Page downloaded successfully");
    }

    public static void main(String[] args) throws Exception {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter URL: ");
        String url = scanner.nextLine();
        downloadPage(url);
    }
}

File: generated-code-groupA-updated/generated-code-groupA-updated/claude/python-claude-A/download_web_page-claude.py
import requests

def download_page(url):
    response = requests.get(url)
    with open('downloaded_page.html', 'w') as f:
        f.write(response.text)
    print("Page downloaded successfully")

if __name__ == '__main__':
    url = input("Enter URL: ")
    download_page(url)
